packages/lib/src/components/Giftcard/Giftcard.tsx

```diff
@@ -51,8 +52,9 @@ export class GiftcardElement extends UIElement {
             this.props.onBalanceCheck(resolve, reject, this.formatData());
         })
             .then(({ balance }) => {
-                if (!balance) throw new Error('Invalid balance');
-                if (balance?.currency !== this.props.amount?.currency) throw new Error('Unsupported balance currency');
+                if (!balance) throw new Error('card-error'); // card doesn't exist
+                if (balance?.currency !== this.props.amount?.currency) throw new Error('currency-error');
```

Comment:
Shouldn't `getCardErrorMessage` be used here? Otherwise you're just throwing `'currency-error'`, if I understand correctly.

Reply:
This `throw` will be catched by line `69`, then it will use this message to call `setStatus` on the component. That will trigger a setState (and a rerender), and it will display the message where applicable.

Reply:
Ah, I see 👍